tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -1672,6 +1672,43 @@ public async Task ValidateGetAccountAsyncWithNullEmptyAccountIdAsync(string acco
                 Assert.IsNull(acc);
             }
         }
+
+        [TestMethod]
+        public async Task ValidateAppTokenProviderAsync()
+        {
+            var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                          .WithAppTokenProvider(async (AppTokenProviderParameters parameters) =>
+                                                          {
+                                                              Assert.IsNotNull(parameters.Scopes);
+                                                              Assert.IsNotNull(parameters.CorrelationId);
+                                                              Assert.IsNotNull(parameters.TenantId);
+                                                              Assert.IsNotNull(parameters.CancellationToken);
+
+                                                              return await Task.Run(() => GetAppTokenProviderResult()).ConfigureAwait(false);
```

Comment:
I think `Task.Run` will dispatch the execcution to a background thread. Not what you need here. It's simpler to just adapt an existing object to a Task result, i.e. 

```csharp
var appTokenProvider = GetAppTokenProviderResult();
return Task.FromResult(appTokenProvider);
```